common/src/main/java/org/corfudb/common/metrics/micrometer/MeterRegistryProvider.java

```diff
@@ -100,6 +106,32 @@ private static synchronized void initLoggingRegistry(Logger logger, Duration log
             addToCompositeRegistry(supplier);
         }
 
+        private static Map<String, String> loadRegistryData() {
+            Map<String, String> map = new HashMap<>();
+            try (InputStream inputStream = MeterRegistryProvider.class.getResourceAsStream(PROVIDED_REGISTRY_DATA_FILE_PATH)) {
+                if (inputStream != null) {
+                    try(BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+                        String line;
+                        while ((line = reader.readLine()) != null) {
+                            if (line.contains("=")) {
+                                String[] split = line.split("=");
+                                if (split.length == 2) {
+                                    map.put(split[0], split[1]);
+                                }
+                            }
+                        }
+                    }
+                }
+                else {
+                    log.warn("Provided file not found: {}. Will return an empty map.", PROVIDED_REGISTRY_DATA_FILE_PATH);
```

Comment:
if the registry.dat is missing then will any metrics be collected at all?

Reply:
If it's missing then the registryProvider.provideRegistry will just create a registry without it. This file is needed for passing metadata. So if some created registry needs to have some metadata for its creation, the protocol is that registry.dat must be located in the classpath.